include/server/request_parser.hpp

```diff
@@ -22,11 +21,18 @@ class RequestParser
   public:
     RequestParser();
 
-    std::tuple<util::tribool, http::compression_type>
+    std::tuple<RequestStatus, http::compression_type>
     parse(http::request &current_request, char *begin, char *end);
 
+    enum class RequestStatus : char
```

Comment:
Move this type declaration before the function declaration. The C++ compiler reads from top to bottom and doesn't know about stuff that is yet to come.